lib/social.ts

```diff
@@ -1,6 +1,21 @@
 import { Story } from '@prisma/client'
 import { storyCite } from '../components/stories/model'
 
-export default function generateSocial(story: Story, contentSize = 150, citeSize = 60) {
-  return `"${story.content.slice(0, contentSize)}" ${storyCite(story).slice(0, citeSize)}`
+export default function generateSocial(story: Story, contentSize = 150, citeSize = 60): string {
+  const { content } = story
+  const cite = storyCite(story).slice(0, citeSize)
+
+  if (content.length < contentSize) {
+    return `"${content}" ${cite}`
+  } else {
+    const storySnippet = content.slice(0, contentSize)
+
+    // Trim the story to the last space
+    const trimmedStorySnippet = storySnippet.substring(
+      0,
+      Math.min(storySnippet.length, storySnippet.lastIndexOf(' '))
+    )
+
+    return `"${trimmedStorySnippet}..." ${storyCite(story).slice(0, citeSize)}`
+  }
```

Comment:
One final nitpick, the else isn't necessary in this case. Removing it makes it slightly easier to read.

```suggestion
  if (content.length < contentSize) {
    return `"${content}" ${cite}`
  }
  const storySnippet = content.slice(0, contentSize)

  // Trim the story to the last space
  const trimmedStorySnippet = storySnippet.substring(
    0,
    Math.min(storySnippet.length, storySnippet.lastIndexOf(' '))
  )

  return `"${trimmedStorySnippet}..." ${storyCite(story).slice(0, citeSize)}`
```